### PR TITLE
Tests/aea configurations

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Working towards full coverage of `aea/configurations`

```
Name                               Stmts   Miss  Cover   Missing
----------------------------------------------------------------
aea/configurations/__init__.py         1      0   100%
aea/configurations/base.py           666      0   100%
aea/configurations/constants.py       77      0   100%
aea/configurations/data_types.py     344      0   100%
aea/configurations/loader.py         166      0   100%
aea/configurations/manager.py        209      0   100%
aea/configurations/pypi.py            94      0   100%
aea/configurations/utils.py           61      0   100%
aea/configurations/validation.py     156      0   100%
----------------------------------------------------------------
TOTAL                               1774      0   100%

```

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by DavidMinarsch** [comment]
> We will need to move `autonomy/configurations/schemas/service_schema.json` to open-aea and create a dummy service.yaml file to attain full coverage of this module

Just create a dummy file. Service is just a dummy here, no assumption should be made that it matches the definition in open autonomy.

File: docs/package_list.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
 | protocol/valory/ledger_api/1.0.0                              | `bafybeih7rhi5zvfvwakx5ifgxsz2cfipeecsh7bm3gnudjxtvhrygpcftq` |
 | connection/fetchai/http_server/0.22.0                         | `bafybeifbdjgzroxywqu5bimvj373yduhnc5n27vxbjf3fqs3mx7s6a5etq` |
 | connection/fetchai/stub/0.21.0                                | `bafybeieqlozydyvdxmjxhqygwq27djecpiftoqwlcpcr4qpotomwnh66yy` |
-| connection/valory/ledger/0.19.0                               | `bafybeic34ll726rlh4bwaeiuzmw5aoiw54ikvdgznlc5q2njv43affylpe` |
```
Comment (angrybayblade) [inline docs/package_list.md:12]: Why the change?

File: tests/test_configurations/test_loader.py [hunk 2]
```diff
@@ -52,6 +52,13 @@ def test_windows_uri_path():
         assert output == f"file:/{'/'.join(path.parts)}/"
 
 
+@pytest.mark.parametrize("aea_config_yaml", Path(ROOT_DIR).rglob("**/aea-config.yaml"))
+def test_load_aea_config(aea_config_yaml):
+    """Test loading aea-config.yaml file"""
+    config_loader = ConfigLoader("aea-config_schema.json", AgentConfig)
+    config_loader.load(open(aea_config_yaml))
```
Comment (angrybayblade) [inline tests/test_configurations/test_loader.py:59]: We should have a negative test case as well

File: docs/package_list.md (hunk 1 above)
Comment (Karrenbelt) [reply docs/package_list.md:12]: not sure, because of changes in loader.py? I never question the new hashes to be quite frank

File: tests/test_configurations/test_loader.py [hunk 3]
```diff
@@ -94,36 +107,53 @@ def test_config_loader_load_service_config(*_mocks):
...
 
-        dummy_obj = MagicMock()
-        dummy_obj.package_type = PackageType.SERVICE
-        dummy_obj.ordered_json = config
+        config_file.touch()
+        with pytest.raises(ValueError, match="Agent configuration file was empty."):
+            config_loader.load(config_file.open("r"))
 
-        config_loader.dump(dummy_obj, config_file.open("w+"))
+        config_loader.dump(dummy_service_config, config_file.open("w+"))
```
Comment (Karrenbelt) [inline tests/test_configurations/test_loader.py:140]: using a non-magic mock allows for the file to be non-empty and improves test coverage

File: tests/test_configurations/test_loader.py [hunk 4]
```diff
@@ -101,27 +101,37 @@ def test_config_loader_load_service_config(*_mocks):
...
+        config_loader_cls.from_json = lambda x: dummy_service_config
         config_loader = ConfigLoader.from_configuration_type(
             PackageType.SERVICE, {PackageType.SERVICE: config_loader_cls}
         )
 
-        dummy_obj = MagicMock()
-        dummy_obj.package_type = PackageType.SERVICE
-        dummy_obj.ordered_json = config
+        config_file.touch()
+        with pytest.raises(ValueError, match="Agent configuration file was empty."):
```
Comment (angrybayblade) [inline tests/test_configurations/test_loader.py:137]: What `Agent configuration file was empty.` error message has to do with the service config test

Comment (Karrenbelt) [reply tests/test_configurations/test_loader.py:137]: well absolutely nothing of course. Some deviant character has copy pasted code it seems!

https://github.com/valory-xyz/open-aea/blob/5a78c4cebe8d61a7321df01fbd1cfe47397a1709/aea/configurations/loader.py#L273-L274
https://github.com/valory-xyz/open-aea/blob/5a78c4cebe8d61a7321df01fbd1cfe47397a1709/aea/configurations/loader.py#L321-L325

File: docs/package_list.md (hunk 1 above)
Comment (angrybayblade) [reply docs/package_list.md:12]: `loader.py` has nothing to with the hashes, please investigate this

File: aea/configurations/loader.py [hunk 5]
```diff
@@ -199,28 +200,27 @@ def load(self, file_pointer: TextIO) -> T:
...
-            return cast(T, self._load_agent_config(file_pointer))
 
-        if self.configuration_class.package_type == PackageType.SERVICE:
-            return cast(T, self._load_service_config(file_pointer))
+        loader: Callable = {
+            PackageType.AGENT: self._load_agent_config,
+            PackageType.SERVICE: self._load_service_config,
+        }.get(self.configuration_class.package_type, self._load_component_config)
 
-        return self._load_component_config(file_pointer)
```
Comment (angrybayblade) [inline aea/configurations/loader.py:208]: Please avoid updating code in test only PRs, open and issue to deal with this later and try to keep the PR strictly for test coverage

File: aea/configurations/loader.py [hunk 6]
```diff
@@ -199,28 +200,27 @@ def load(self, file_pointer: TextIO) -> T:
...
         :param file_pointer: the file pointer to the configuration file
         """
-        if self.configuration_class.package_type == PackageType.AGENT:
-            self._dump_agent_config(cast(AgentConfig, configuration), file_pointer)
-        elif self.configuration_class.package_type == PackageType.SERVICE:
-            self._dump_service_config(
-                cast(PackageConfiguration, configuration), file_pointer
-            )
-        else:
-            self._dump_component_config(configuration, file_pointer)
```
Comment (angrybayblade) [inline aea/configurations/loader.py:223]: Same as above

File: tests/test_configurations/test_loader.py (hunk 4 above)
Comment (angrybayblade) [reply tests/test_configurations/test_loader.py:137]: Please update the error message

File: tests/test_cli/test_utils/test_utils.py [hunk 7]
```diff
@@ -208,6 +208,7 @@ def _raise_file_not_found_error(*args):
     raise FileNotFoundError()
 
 
+@pytest.skip(reason="flaky test addressed in PR #369")
```
Comment (angrybayblade) [inline tests/test_cli/test_utils/test_utils.py:211]:
```suggestion
@pytest.mark.skip(reason="flaky test addressed in PR #369")
```

File: tests/test_configurations/test_loader.py (hunk 2 above)
Comment (Karrenbelt) [reply tests/test_configurations/test_loader.py]: the negative, as in empty, is covered in tests as well

File: aea/configurations/loader.py (hunk 5 above)
Comment (DavidMinarsch) [reply aea/configurations/loader.py:208]: Agreed!

File: aea/configurations/loader.py (hunk 6 above)
Comment (DavidMinarsch) [reply aea/configurations/loader.py:223]: Agreed!

**Review by DavidMinarsch — APPROVED** [review]
